File: SecretSanta.cpp
#include "SecretSanta.h"
#include <string>

SecretSanta::SecretSanta() {
	this->name = "";
	this->wishList = "";
}
SecretSanta::SecretSanta(std::string name) {
	this->name = name;
	this->wishList = "";
}
SecretSanta::SecretSanta(std::string name, std::string wishList) {
	this->name = name;
	this->wishList = wishList;
}
	
std::string SecretSanta::getName() const {
	return this->name;
}
	
std::string SecretSanta::getWishList() const {
	return this->wishList;
}
	
void SecretSanta::addToWishList(std::string item) {
	if (this->wishList = "")
		this->wishList += item;
	else
		this->wishList = this->wishList + ", " + item;
}
	
void SecretSanta::removeFromWishList(std::string item) {
	std::string::size_type i = this->wishList.find(item);
	if (i != std::string::npos)
   		this->wishList.erase(i, item.length());
}
	
SecretSanta SecretSanta::getRecipient() const {
	return this->recipient;
}
	
std::string SecretSanta::getRecipientWishlist() const {
	return this->recipient.wishList;
}

File: SecretSanta.java
import java.util.ArrayList;

public class SecretSanta {
	private String name;
	private WishList santaWishList;
	private SecretSanta recipient;
	
	class WishList {
		private ArrayList<String> list;

		WishList() {
			list = new ArrayList<String>();
		}

		String outputList() {
			return list.toString();
		}

		void addToWishList(String item){
			list.add(item);
		}
		void removeFromWishList(String item){
			list.remove(item);
		}
	}
	SecretSanta(){
		name = "";
		santaWishList = new WishList();
	}
	SecretSanta(String santaName){
		name = santaName;
		santaWishList = new WishList();
	}
	SecretSanta(String santaName, String item){
		name = santaName;
		santaWishList = new WishList();
		santaWishList.addToWishList(item);
	}
	String getName(){
		return name;
	}
	WishList getWishList(){
		return santaWishList;
	}
	
	SecretSanta getRecipient(){
		return recipient;
	}
	void setRecipient(SecretSanta recip){
		recipient = recip;
	}
	String getRecipientWishList(){
		return recipient.getWishList().toString();
	}
	String output() {
		return "Name: " + name + "\n" + "Wishlist: " + santaWishList.toString();
	}
}
